src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -436,13 +436,15 @@ private boolean quickSetSolid(final TransactionViewModel transactionViewModel) t
      * @throws Exception if we encounter an error while requesting a transaction
      */
     private boolean checkApproovee(TransactionViewModel approovee) throws Exception {
-        if(approovee.getType() == PREFILLED_SLOT) {
-            transactionRequester.requestTransaction(approovee.getHash(), false);
-            return false;
-        }
         if(snapshotProvider.getInitialSnapshot().hasSolidEntryPoint(approovee.getHash())) {
             return true;
         }
+        if(approovee.getType() == PREFILLED_SLOT) {
+            // don't solidify from the bottom until cuckoo filters can identify where we deleted -> otherwise we will
+            // continue requesting old transactions forever
+            //transactionRequester.requestTransaction(approovee.getHash(), false);
```

Comment:
this change was made to prevent false positives - transactions that we think are missing, but have deliberately been deleted.

but it seems this will create false negatives - where actual missing transactions (small holes) will not be requested and thought of as deleted.

Reply:
That is correct @alon-e but they will be requested by the checkSolidity calls from the top. So as long as we don't have a way to determine where to stop the solidification (the cuckoo filters PR prepared and merged), this will break nodes that have local snapshots enabled. If we solidify from the top we stop whenever we reach a solid entry point.

But if we happen to receive a transaction that is "below" a solid entry point - this logic will continue requesting everything below it until we reach the genesis.